src/main/java/eventprocessing/models/BucketManager.java

```diff
@@ -0,0 +1,88 @@
+package eventprocessing.models;
+
+import eventprocessing.GlobalConstants;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BucketManager {
```

Comment:
This is another class where it would be useful to think about how the data and operations could be combined.

This class is called `BucketManager` but it doesn't do a lot of managing - it gets all of its instructions from outside the class, in `Main`.

How could you refactor the app so that `BucketManager` took responsibility for deciding when to roll over to the next bucket? It would move a lot of complicated logic out of `Main` and simplify the API of this class.